### PR TITLE
New Feature: Move The Language Selector [ en fr ] To The Top-right Corner Of The Page.

### PR DESCRIPTION
**Pull request by 0xull**
Patch: Changed the language selector [ en fr ] component to the top-right corner of the page: on the desktop, it's next to the edit button; on the phone that's next to the hamburger menu. This follows the left-to-right design layout that suggests the language selector should be in the top-right corner of the page, that's where people intuitively would look for it

Test: The responsiveness and consistency of the component across all screen resolution, and also ascertain how intuitive the change truly is to design layout.

Previous:
ocaml-prev

Recent:
ocaml-recent

### CODE REVIEW

**Comment by 0xull** [comment]
This PR is made in response to the issue the last PR made to the `staging branch` at [#1326](https://github.com/ocaml/ocaml.org/pull/1326) had.

**Comment by patricoferris** [comment]
HI @IkehAkinyemi, thanks for this PR. A few suggestions: 

 - Typically for pull requests it is nearly always best to keep them narrow in scope. If the title of the PR is about `UI Fixes` then adding commits which update the on-boarding documentation is perhaps not the right place for those. It makes it harder to review and if people come back to this PR in the future (say it broke something and wanted to revert the merge) your other work would likely be taken out at the same time.
 - I had a look on staging.ocaml.org at the proposed changes and I'm not entirely convinced by them just yet. Imo, I'm not sure the top-right corner is much clearer for users that the language selector would be there. In addition to this the orange (non-selected languages) have a slight "halo" around them which makes them harder to read (see this [accessibility check](https://wave.webaim.org/report#/https://staging.ocaml.org/)).
 
Is there an associated issue with this PR (for the UI not the documentation part)? If not, going through an issue first is usually a better option than submitting a PR for a change without one. We can discuss the problem before implementing the solution which saves time and work in the long run.

**Comment by 0xull** [comment]
> Typically for pull requests it is nearly always best to keep them narrow in scope. If the title of the PR is about UI Fixes then adding commits that update the onboarding documentation is perhaps not the right place for those. It makes it harder to review and if people come back to this PR in the future (say it broke something and wanted to revert the merge) your other work would likely be taken out at the same time.

I agree with you, the scope of the title, UI fixes, doesn't align with the PR been made. I will change that to better narrow down the scope of the patch.

> Is there an associated issue with this PR (for the UI, not the documentation part)? If not, going through an issue first is usually a better option than submitting a PR for a change without one. We can discuss the problem before implementing the solution which saves time and work in the long run.

It's a mistake on my side not to have referenced the link to the issue I submitted the PR for. The PR is made for the issue raised by @pitag-ha at [#1305 ](https://github.com/ocaml/ocaml.org/issues/1305). And for the most part, it was registered and the reasons were well enough discussed there.

> I had a look on staging.ocaml.org at the proposed changes and I'm not entirely convinced by them just yet. Imo, I'm not sure the top-right corner is much clearer for users that the language selector would be there. In addition to this the orange (non-selected languages) have a slight "halo" around them which makes them harder to read (see this accessibility check).

I agree with you, the color contrast of the `color: #a20505;` on the `[ fr ]` language selector isn't a good design practice but I wasn't sure what choice of color would be the best fitting for it. The orange variant is Ocaml color identity. But I would love to hear your suggestion on what color will be well-fitting for the `[ fr ]` language selector.

**Comment by patricoferris** [comment]
Ah no that's totally my bad, there's so many issues and PRs it's quite hard to keep track. Sorry if I came across too abrupt, I'm communicating with lots of people across many forums and I end up typing fast with less thought, that's no accuse so apologies. 

> I agree with you, the scope of the title, UI fixes, doesn't align with the PR been made. I will change that to better narrow down the scope of the patch.

These changes are great so please do open a PR with them.

> I agree with you, the color contrast of the color: #a20505; on the [ fr ] language selector isn't a good design practice but I wasn't sure what choice of color would be the best fitting for it. The orange variant is Ocaml color identity. But I would love to hear your suggestion on what color will be well-fitting for the [ fr ] language selector

I'm really not sure 😅  so long as it is clear from a contrast level and also clear to indicate what language someone is on. If it is also similar to the orange colour scheme that would be good too. [Google Lighthouse has a good tool for testing contrast levels](https://webdesign.tutsplus.com/articles/how-to-use-the-contrast-checker-in-chrome-devtools--cms-31504) which might be useful.

File: site/css/ocamlorg.css [hunk 1]
```diff
@@ -522,6 +539,12 @@ ul.translations:after { content: "]"; }
     margin-right: 0px;
   }
 }
+@media (min-width: 980px)and (max-width: 1058px) {
+  #searchform, #search {
+    margin-right: 1.8rem;
+    width: 125px !important;
+  }
+}
```
Comment (pitag-ha) [inline site/css/ocamlorg.css:547]: This fixes a [different problem](https://github.com/ocaml/ocaml.org/issues/1320), doesn't it? For future reference: in general it is a good practice to keep things separate (separate commits; separate PR's). But nice that this is fixed!

**Review by pitag-ha — COMMENTED** [review]
Thanks, this is nice contribution! In addition to what Patrick said, I've also found the following problem on the tutorials page:

Screenshot from 2021-03-31 18-39-32

That's a tough one, but can you try to figure out how to make sure that the language selector doesn't overlap with the search bar?

**Pull request closed without merging**